src/python/command_line.py

```diff
@@ -0,0 +1,69 @@
+import subprocess
+from tempfile import TemporaryDirectory
+from typing import List
+import json
+
+
+# Function that calls vehicle - it takes the options
+# TODO: check if it's None or else
+def call_vehicle(args:List[str]) -> None:
+    command = ['vehicle'] + args
+    # print(' '.join(command))
+    result = subprocess.run(command, capture_output=True)
+    if result.returncode != 0:
+        raise Exception('Error during specification compilation: ' + result.stderr.decode('UTF-8'))
+
+
+def load_json(path_to_json:str) -> dict:
+    with open(path_to_json) as f:
+        json_dict = json.load(f)
+    return json_dict
+
+
+def call_vehicle_to_generate_loss_json(specification:str, function_name:str) -> dict:
+    with TemporaryDirectory() as path_to_json_directory:
+        path_to_json = path_to_json_directory + 'loss_function.json'
+        args =  ['compile',
+                '--target', 'LossFunction', 
+                '--specification', specification, 
+                '--outputFile', path_to_json]
+                #'--property', function_name]
+        call_vehicle(args)
+        loss_function_json = load_json(path_to_json)
+    return loss_function_json
+
+
+def make_network_argument(networks:dict[str, str]) -> List[str]:
+    network_list = []
+    for name, location in networks.items():
+        network_list.append('--network')
+        network_list.append(name + ':' + location)
+    return network_list
+
+
+def make_dataset_argument(datasets:dict[str, str]) -> List[str]:
+    dataset_list = []
+    for name, location in datasets.items():
+        dataset_list.append('--network')
```

Comment:
This and `make_parameter_argument` look wrong. It should be `--dataset` not `--network` right? 

More generally, all three methods are very similar. Maybe it should just be one generic function `make_resource_arguments(resources:dict[str, str], arg_name:str)` where `arg_name` is the name of the command line argument?

Reply:
Yep, combined them into one.